### PR TITLE
feat(eslint): migrate sdk client packages

### PR DESCRIPTION
**Pull request by matthieu-crouzet**

### CODE REVIEW

File: packages/@ama-sdk/client-angular/src/api-angular-client.ts [hunk 1]
```diff
@@ -144,13 +149,14 @@ export class ApiAngularClient implements ApiClient {
...
           next: (res) => data = res,
+          // eslint-disable-next-line @typescript-eslint/prefer-promise-reject-errors -- subscription forwards the error from the httpRequest to asyncResponse promise
           error: (err) => reject(err),
           complete: () => resolve(data)
         });
         metadataSignal?.throwIfAborted();
         metadataSignal?.addEventListener('abort', () => {
           subscription.unsubscribe();
-          reject(metadataSignal.reason);
+          reject(metadataSignal.reason instanceof Error ? metadataSignal.reason : new Error(metadataSignal.reason.toString()));
```
Comment (sdo-1A) [inline packages/@ama-sdk/client-angular/src/api-angular-client.ts:159]: shouldn't the format look something like this?
```suggestion
          reject(metadataSignal.reason instanceof Error 
            ? metadataSignal.reason 
            : new Error(metadataSignal.reason.toString()));
```

File: packages/@ama-sdk/client-angular/src/api-angular-client.ts [hunk 2]
```diff
@@ -159,10 +165,10 @@ export class ApiAngularClient implements ApiClient {
       exception = new EmptyResponseError(e.message || 'Fail to Fetch', undefined, { apiName, operationId, url, origin });
     }
 
-    // eslint-disable-next-line no-console
+    // eslint-disable-next-line no-console -- `console.error` is supposed to be the default value if the `options` argument is not provided, can be removed in Otter v12.
     const reviver = getResponseReviver(revivers, response, operationId, { disableFallback: this.options.disableFallback, log: console.error });
```
Comment (cpaulve-1A) [inline packages/@ama-sdk/client-angular/src/api-angular-client.ts:169]: Careful if we pass the logger to communicate with the teams before: there are a lot of logs that can be hidden behind this one

File: packages/@ama-sdk/client-angular/package.json [hunk 3]
```diff
@@ -80,28 +80,35 @@
     "@angular-eslint/eslint-plugin": "~18.3.0",
```
Comment (sdo-1A) [inline packages/@ama-sdk/client-angular/package.json:80]: shouldn't this dependency be removed?

File: packages/@ama-sdk/client-fetch/src/api-fetch-client.ts [hunk 4]
```diff
@@ -145,22 +144,18 @@ export class ApiFetchClient implements ApiClient {
 
       body = await response.text();
     } catch (e: any) {
-      if (e instanceof CanceledCallError) {
-        exception = e;
-      } else {
-        exception = new EmptyResponseError(e.message || 'Fail to Fetch', undefined, {apiName, operationId, url, origin});
-      }
+      exception = e instanceof CanceledCallError ? e : new EmptyResponseError(e.message || 'Fail to Fetch', undefined, { apiName, operationId, url, origin });
```
Comment (sdo-1A) [inline packages/@ama-sdk/client-fetch/src/api-fetch-client.ts:147]: format?
```suggestion
      exception = e instanceof CanceledCallError 
        ? e 
        : new EmptyResponseError(e.message || 'Fail to Fetch', undefined, { apiName, operationId, url, origin });
```

File: packages/@ama-sdk/client-fetch/src/plugins/perf-metric/perf-metric.fetch.ts [hunk 5]
```diff
@@ -120,7 +126,7 @@ export class PerformanceMetricPlugin implements FetchPlugin {
 
   constructor(options?: Partial<PerformanceMetricOptions>) {
     this.getPerformanceTag = options?.getPerformanceTag || this.getPerformanceTag;
-    this.performance = options?.performance || (typeof window !== 'undefined' ? window.performance : undefined);
+    this.performance = options?.performance || (typeof window === 'undefined' ? undefined : window.performance);
```
Comment (cpaulve-1A) [inline packages/@ama-sdk/client-fetch/src/plugins/perf-metric/perf-metric.fetch.ts:129]:
```suggestion
    this.performance = options?.performance || window?.performance;
```

File: packages/@ama-sdk/client-angular/src/api-angular-client.ts (hunk 2 above)
Comment (matthieu-crouzet) [reply packages/@ama-sdk/client-angular/src/api-angular-client.ts:173]: after discussion text is okay

**Review by sdo-1A — APPROVED** [review]

**Review by cpaulve-1A — APPROVED** [review]

**Pull request merged**